### PR TITLE
[CRIMAPP-983, 984, 985] Add conditional self employed screens

### PR DESCRIPTION
**Pull request by hiboabd**
## Description of change
Adds conditional partnership/director stakeholder business questions

## Link to relevant ticket
[CRIMAPP-983](https://dsdmoj.atlassian.net/browse/CRIMAPP-983)
[CRIMAPP-984](https://dsdmoj.atlassian.net/browse/CRIMAPP-984)
[CRIMAPP-985](https://dsdmoj.atlassian.net/browse/CRIMAPP-985)

### After changes:

Screenshot 2024-07-09 at 12 37 17
Screenshot 2024-07-09 at 12 37 09
Screenshot 2024-07-09 at 12 37 00
Screenshot 2024-07-09 at 12 36 48
Screenshot 2024-07-09 at 12 37 50
Screenshot 2024-07-09 at 12 37 32

## How to manually test the feature
Start an application, select self employed for employment status, test the self employed, partnership and director or stakeholder business flows

[CRIMAPP-983]: https://dsdmoj.atlassian.net/browse/CRIMAPP-983?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMAPP-984]: https://dsdmoj.atlassian.net/browse/CRIMAPP-984?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMAPP-985]: https://dsdmoj.atlassian.net/browse/CRIMAPP-985?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/services/decisions/self_employed_income_decision_tree.rb [hunk 1]
```diff
@@ -17,7 +17,13 @@ def destination # rubocop:disable Metrics/MethodLength, Metrics/CyclomaticComple
       when :business_employees
         edit(:business_financials, business_id:)
       when :business_financials
-        edit(:businesses_summary, business_id:)
+        after_business_financials
+      when :business_salary_or_remuneration
+        edit(:business_total_income_share_sales, business_id: form_object.record)
```
Comment (timpeat) [inline app/services/decisions/self_employed_income_decision_tree.rb:22]: business_id is set as a method below so form_object.record is not required

File: app/services/decisions/self_employed_income_decision_tree.rb [hunk 2]
```diff
@@ -17,7 +17,13 @@ def destination # rubocop:disable Metrics/MethodLength, Metrics/CyclomaticComple
...
+      when :business_total_income_share_sales
+        edit(:business_percentage_profit_share, business_id: form_object.record)
```
Comment (timpeat) [inline app/services/decisions/self_employed_income_decision_tree.rb:24]: ditto

File: app/services/decisions/self_employed_income_decision_tree.rb [hunk 3]
```diff
@@ -17,7 +17,13 @@ def destination # rubocop:disable Metrics/MethodLength, Metrics/CyclomaticComple
...
+      when :business_percentage_profit_share
+        edit(:businesses_summary, business_id: form_object.record)
```
Comment (timpeat) [inline app/services/decisions/self_employed_income_decision_tree.rb:26]: ditto

**Review by timpeat — APPROVED** [review]

File: app/services/decisions/self_employed_income_decision_tree.rb (hunk 1 above)
Comment (hiboabd) [reply app/services/decisions/self_employed_income_decision_tree.rb:22]: how did i miss that 🤦‍♀️

**Pull request merged**